records/filter.record.ts

```diff
@@ -0,0 +1,141 @@
+type AllRecordsStudentResults = [{totalCount:number}[],FieldPacket[]];
+
+export class FilterRecord implements FilterEntity{
+    remoteWork: boolean|string;
+    inOffice: boolean|string;
+    employmentContract: boolean|string;
+    b2b: boolean|string;
+    mandateContract: boolean|string;
+    workContract: boolean|string;
+    min: string|number;
+    max: string|number;
+    canTakeApprenticeship: boolean|null|string;
+    monthsOfCommercialExp: string|null;
+    courseCompletion: string|number;
+    courseEngagement: string|number;
+    projectDegree: string|number;
+    teamProjectDegree: string|number;
+    page: string|number;
+    rowsPerPage : string|number;
+    hrId?: string;
+
+    constructor(obj:FilterEntity) {
+        //@TODO można dodać walidacje danych
+        this.remoteWork = obj.remoteWork;
+        this.inOffice = obj.inOffice;
+        this.employmentContract = obj.employmentContract;
+        this.b2b = obj.b2b;
+        this.mandateContract = obj.mandateContract;
+        this.workContract = obj.workContract;
+        this.min = obj.min;
+        this.max = obj.max;
+        this.canTakeApprenticeship = obj.canTakeApprenticeship;
+        this.monthsOfCommercialExp = obj.monthsOfCommercialExp;
+        this.courseCompletion = obj.courseCompletion;
+        this.courseEngagement = obj.courseEngagement;
+        this.projectDegree = obj.projectDegree;
+        this.teamProjectDegree = obj.teamProjectDegree;
+        this.page = obj.page;
+        this.rowsPerPage = obj.rowsPerPage;
+        this.hrId=obj.hrId;
+    }
+
+    change(){
+        let query = "";
+        if(this.remoteWork === 'true' || this.inOffice === 'true'){
+            query +=  "`expectedTypeWork` IN (:remoteWork, :inOffice)";
+        }
+        if(this.employmentContract === true || this.b2b === true || this.mandateContract === true || this.workContract === true){
+            if(query !== ""){
+                query += " AND ";
+            }
+            query += "`expectedContractType` IN (:employmentContract, :b2b,:mandateContract,:workContract)"
+        }
+        if(this.canTakeApprenticeship !== 'null'){
+            if(query !== ""){
+                query += " AND ";
+            }
+            query += "`canTakeApprenticeship` = :canTakeApprenticeship"
+        }
+        if(query !== ""){
+            query += " AND ";
+        }
+        this.remoteWork = this.remoteWork === 'true' ? '2' : '9';
+        this.inOffice = this.inOffice === 'true' ? '1' : '9';
+        this.employmentContract = this.employmentContract === 'true' ? '1' : '9';
+        this.b2b = this.b2b === `true` ? '2' : '9';
+        this.mandateContract = this.mandateContract === 'true' ? '3' : '9';
+        this.workContract = this.workContract === 'true' ? '4' : '9';
+        this.page = Number(this.page);
+        this.rowsPerPage = Number(this.rowsPerPage);
+        this.page = this.page * this.rowsPerPage;
+        return query;
```

Comment:
Rozumiem koncepcję i szanuję, że nie ma tu (jeszcze) potencjalnie SQLInjection, choć miało taki potencjał na pierwszy rzut oka :)
Natomiast mocno zastanowiłbym się nad wykorzystaniem jakiegoś ORM-a albo chociaż QueryBuildera, bo jest to nieco nieczytelne.